src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelSchema.scala

```diff
@@ -59,3 +43,41 @@ case class EntityMetadata(
 )
 
 case class EntityModel(schema : Map[String, EntityMetadata]) //entity name -> stuff about it
+
+object FlexibleModelSchema {
+  def memberTypeFromEntityType(entityType: String): Try[String] = {
+    val memberType = EntityTypes.types.get(entityType) match {
+      case Some(schema) => schema.memberType
+      case None => if (entityType.endsWith("_set"))
+        Some(entityType.replace("_set", "")) else
```

Comment:
Nit: instead of `else None`, I think a 3rd `case` would read a little bit better:
```
      case Some(schema) => schema.memberType
      case None if entityType.endsWith("_set") =>
        Some(entityType.replace("_set", ""))
      case _ => None
```